geoh5py/workspace/workspace.py

```diff
@@ -1228,6 +1224,25 @@ def open(self, mode: str | None = None) -> Workspace:
 
         return self
 
+    def register(self, entity: Entity | EntityType | PropertyGroup):
+        """
+        Register an entity to the workspace based on its type.
+
+        :param entity: The entity to be registered.
+        """
+        if isinstance(entity, EntityType):
+            self._register_type(entity)
+        elif isinstance(entity, Group):
+            self._register_group(entity)
+        elif isinstance(entity, Data):
+            self._register_data(entity)
+        elif isinstance(entity, ObjectBase):
+            self._register_object(entity)
+        elif isinstance(entity, PropertyGroup):
+            self.register_property_group(entity)
```

Comment:
**thought**: I see `PropertyGroup` does not inherit from `Entity` and calls directly `self.parent.workspace.register_property_group(self)`

Thus it will not go through this `register()` function. For consistency, `PropertyGroup __init__()` could call `workspace.register()`. and possibly the code of `register_property_group()` is moved to `_register_property_group()`

I am also curious: why this special case with PropertyGroup?
```
        if not property_group.on_file:
            self.add_or_update_property_group(property_group)
```

Would it make sense that `PropertyGroup` inherits from `Entity`?

Reply:
A bunch of methods are not shared between entity and PropertyGroup.
Still, it might be possible to create some superclass over entity. 

Dom, what do you think?

Reply:
Property groups are different since they are only registered under the parent object, and not under specific h5 groups. I think it was just a typo here with accessing a non-private method? This issue is already getting way out of scope, I would avoid touching property_group at this stage.

Reply:
I agree changing intheritence of PropertyGroup is out of scope. I just want ed to raise the question.
Thanks @domfournier for the clarification.